product_import_ubl/wizard/product_import.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2022 Camptocamp
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from lxml import etree
+
+from odoo import api, models
+
+CATALOGUE_NS = "urn:oasis:names:specification:ubl:schema:xsd:Catalogue-2"
+CATALOGUE_TAG = "{%s}Catalogue" % CATALOGUE_NS
+
+
+class XPathGetter(object):
+    _missing = etree.Element("Missing")
+
+    def __init__(self, element, namespaces):
+        self._xpath = element.xpath
+        self._ns = namespaces
+
+    def xpath(self, path):
+        # Return a list of elements
+        items = self._xpath(path, namespaces=self._ns)
+        return items
+
+    def xpath_get(self, path):
+        # Return 1 element
+        items = self._xpath(path, namespaces=self._ns)
+        return items[0] if items else self._missing
+
+    def xpath_text(self, path):
+        # Return text or False
+        items = self._xpath(path, namespaces=self._ns)
+        return items and items[0].text or False
+
+    get = xpath_get
+    text = xpath_text
+
+
+class ProductImport(models.TransientModel):
+    _name = "product.import"
+    _inherit = ["product.import", "base.ubl"]
+
+    @api.model
+    def parse_xml_catalogue(self, xml_root, detect_doc_type=False):
+        if xml_root.tag == CATALOGUE_TAG:
+            if detect_doc_type:
+                return "catalogue"
+            return self.parse_ubl_catalogue(xml_root)
+        return super().parse_xml_catalogue(xml_root, detect_doc_type=detect_doc_type)
+
+    @api.model
+    def parse_ubl_catalogue_line(self, line, ns):
+        xline = XPathGetter(line, ns)
+
+        # Get product basic info:
+        #  - code: cac:Item/cac:SellersItemIdentification/cbc:ID
+        #  - barcode: cac:Item/cac:StandardItemIdentification/cbc:ID[@schemeID='GTIN']
+        product_vals = self.ubl_parse_product(line, ns)
+
+        # Fill available data
+        ele_price = xline.get(
+            "cac:RequiredItemLocationQuantity/cac:Price/cbc:PriceAmount"
+        )
+        currency = ele_price.attrib.get("currencyID") or False
+        min_qty = float(
```

Comment:
is this float call safe? No risk to get a bad formatted value? Eg: comma instead of dot?

Reply:
It's as safe as we can ... usually it's even integer in UBL, but the Odoo field is float.